docker-java/src/test/java/com/github/dockerjava/cmd/StatsCmdIT.java

```diff
@@ -37,23 +36,52 @@ public void testStatsStreaming() throws InterruptedException, IOException {
 
         dockerRule.getClient().startContainerCmd(container.getId()).exec();
 
-        StatsCallbackTest statsCallback = dockerRule.getClient().statsCmd(container.getId()).exec(
-                new StatsCallbackTest(countDownLatch));
+        Boolean gotStats;
+        try (StatsCallbackTest statsCallback = dockerRule.getClient().statsCmd(container.getId()).exec(
+                new StatsCallbackTest(countDownLatch))) {
+            countDownLatch.await(3, TimeUnit.SECONDS);
+            gotStats = statsCallback.gotStats();
+
+            LOG.info("Stop stats collection");
+        }
+
+        LOG.info("Stopping container");
+        dockerRule.getClient().stopContainerCmd(container.getId()).exec();
+        dockerRule.getClient().removeContainerCmd(container.getId()).exec();
+
+        LOG.info("Completed test");
+        assertTrue("Expected true", gotStats);
+    }
+
+    @Test
+    public void testStatsNoStreaming() throws InterruptedException, IOException {
+        CountDownLatch countDownLatch = new CountDownLatch(NUM_STATS);
+
+        String containerName = "generated_" + new SecureRandom().nextInt();
+
+        CreateContainerResponse container = dockerRule.getClient().createContainerCmd("busybox").withCmd("top")
+                .withName(containerName).exec();
+        LOG.info("Created container {}", container.toString());
+        assertThat(container.getId(), not(isEmptyString()));
```

Comment:
this is tested in `CreateContainerCmdIT`, no need to log/assert it

Reply:
Removed assertions and logs.